src/border.c

```diff
@@ -451,7 +451,17 @@ void DrawBorderHelper(const ClientNode *np)
       if(np->name && np->name[0] && point.x < point.y) {
          unsigned titleWidth = point.y - point.x;
          const int sheight = GetStringHeight(FONT_BORDER);
-         const int textWidth = GetStringWidth(FONT_BORDER, np->name);
+         const int buffSize = (MAX_TITLE_LENGTH+1)*sizeof(char);
+         char *titleBuffer = calloc(MAX_TITLE_LENGTH+1,sizeof(char));
```

Comment:
What's the reason behind a fixed length allocation? I suspect 512 is probably enough for most reasonable use-cases, but then the `sprintf`s should probably be switched to `snprintf`s. I think I'd be more comfortable just allocating the right amount though, which is maybe a little error prone, but I think this would work:

```c
const char *titleBuffer;
if(settings.showMachineName) {
   /* Space for 2 delimiters, space, terminator, and strings */
   const size_t bufSize = strlen(np->name) + strlen(np->machineName) + 4;
   titleBuffer = Allocate(bufSize);
   sprintf(titleBuffer, "%s %c%s%c", np->name,
           settings.machineNameDelimiters[0], np->machineName,
           settings.machineNameDelimiters[1]);
} else {
   titleBuffer = CopyString(np->name);
}
...
Release(titleBuffer);
```

Also, JWM has macros to wrap malloc/realloc/free (Allocate/Reallocate/Release), which I try to use to make it easy to check for memory leaks and buffer overruns.

Reply:
I will make the suggested changes and remove the MAX_TITLE_LENGTH define